app/tx_test.go

```diff
@@ -914,52 +918,213 @@ func TestChangeParamsSimpleTx(t *testing.T) {
+	codec.UpgradeHeight = 2                               // Height at which codec was upgraded from amino to proto
+	codec.UpgradeFeatureMap[codec.BlockSizeModifyKey] = 3 // Height at which to enable block size upgrades
+	_ = memCodecMod(true)
+	resetTestACL()
+
+	// Prepare the test network
+	_, kb, cleanup := NewInMemoryTendermintNodeProto(t, oneAppTwoNodeGenesis())
+	defer cleanup()
+
+	// Get the address of the ACL owner (i.e. the DAO)
+	cb, err := kb.GetCoinbase()
+	assert.Nil(t, err)
+	daoAddr := cb.GetAddress()
+
+	// Subscribe to new events
+	_, _, newBlockEventChan := subscribeTo(t, tmTypes.EventNewBlock)
+	memCli, stopCli, txEventChan := subscribeTo(t, tmTypes.EventTx)
+	defer stopCli()
+
+	<-newBlockEventChan // Wait for block 1
+	<-newBlockEventChan // Wait for block 2
+
+	// Before activation, the parameter does not exist and should be an empty string
+	queryRes, err := PCA.QueryParam(PCA.LastBlockHeight(), blockSizeKey)
+	assert.Nil(t, err)
+	assert.Equal(t, "", queryRes.Value)
+
+	<-newBlockEventChan // Wait for block 3
+
+	// After activation, the parameter should be the default value
+	queryRes, err = PCA.QueryParam(PCA.LastBlockHeight(), blockSizeKey)
+	assert.Nil(t, err)
+	assert.Equal(t, "4000000", queryRes.Value)
+	assert.Equal(t, strconv.Itoa(int(pocketTypes.DefaultBlockByteSize)), queryRes.Value)
+
+	// Changing the parameter after activation
+	tx, err := gov.ChangeParamsTx(memCodec(), memCli, kb, daoAddr, blockSizeKey, newBlockSize, "test", 10000, false)
+	assert.Nil(t, err)
+	assert.NotNil(t, tx)
+	<-txEventChan // wait for change to take affect
+
+	// Parameter should be updated
+	queryRes, err = PCA.QueryParam(PCA.LastBlockHeight(), blockSizeKey)
+	assert.Nil(t, err)
+	assert.Equal(t, newBlockSize, queryRes.Value)
+
+	<-newBlockEventChan // Wait for block 4
+
+	// Verify the parameter change is stable
+	queryRes, err = PCA.QueryParam(PCA.LastBlockHeight(), blockSizeKey)
+	assert.Nil(t, err)
+	assert.Equal(t, newBlockSize, queryRes.Value)
+
+	<-newBlockEventChan // Wait for block 5
+
+	// Verify the parameter change is stable
+	queryRes, err = PCA.QueryParam(PCA.LastBlockHeight(), blockSizeKey)
+	assert.Nil(t, err)
+	assert.Equal(t, newBlockSize, queryRes.Value)
+}
+
+// NOTE: This is a hacky test prone to race conditions and global variable management to account for lack of
+// fine-grained control of Tendermint's consensus engine (which runs in the background).
+//
+// How it works:
+// 1. Configure the network & initialize block size updates
+// 2. Start a background process that keeps sending transactions
+// 3. Start a background process that doubles the block size every block
+// 4. Repeat (2) & (3) N times
+//
+// VERIFICATION: The assertions in the test do minimal verification but evaluating the logs provides an additional guarantee.
+// 1. Verify that smaller block sizes cannot be completely filled
```

Comment:
Are these scenarios you are suggesting to look at when evaluating the logs? I don't see the test(s) for these / not sure how plausible it is for a test like this without localnet/testnet testing

Reply:
<img width="728" alt="Screenshot 2023-04-03 at 6 51 34 PM" src="https://user-images.githubusercontent.com/1892194/229665886-bc8ca957-73c6-45da-8769-a3d1d2fdc28b.png">

> I don't see the test(s) for these / not sure how plausible it is for a test like this without localnet/testnet testing

Yea, the instructions for it are in progress and coming in a follow up PR. Highlighted it in the next steps in the description.

> Are these scenarios you are suggesting to look at when evaluating the logs?

Look at the `The output of the test is as follows:` in the GH description.

Will improve on the assertions.